conda_smithy/linter/hints.py

```diff
@@ -232,3 +232,29 @@ def hint_pip_no_build_backend(host_or_build_section, package_name, hints):
                 "If your recipe has built with only `pip` in the `host` section in the past, you likely should "
                 "add `setuptools` to the `host` section of your recipe."
             )
+
+
+def hint_noarch_python_use_python_min(
+    host_reqs, run_reqs, test_reqs, outputs_section, noarch_value, hints
+):
+    if noarch_value == "python" and not outputs_section:
+        for section_name, syntax, reqs in [
+            ("host", "python {{ python_min }}.*", host_reqs),
+            ("run", "python >={{ python_min }}", run_reqs),
+            ("test.requires", "python ={{ python_min }}", test_reqs),
+        ]:
+            for req in reqs:
+                if (
+                    req.strip().split()[0] == "python"
+                    and req != "python"
+                    and syntax in req
+                ):
+                    break
+            else:
+                hints.append(
+                    f"noarch: python recipes should almost always follow the syntax in "
+                    f"[CFEP-25](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python). "
```

Comment:
Is this supposed to be a link to CFEP-25?
```suggestion
                    f"[`noarch: python`](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) recipes should almost always follow the syntax in "
                    f"[CFEP-25](https://github.com/conda-forge/cfep/blob/main/cfep-25.md). "
```

Reply:
It was supposed to be a link to the docs. I can change the link text.

Reply:
```suggestion
                    f"noarch: python recipes should almost always follow the syntax in "
                    f"our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python). "
```